UITests/Common/BookmarksUtilites.swift

```diff
@@ -0,0 +1,96 @@
+//
+//  BookmarksUtilites.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+
+class BookmarkUtilities {
+
+    /// Reset the bookmarks so we can rely on a single bookmark's existence
+    static func resetBookmarks(app: XCUIApplication, resetMenuItem: XCUIElement) {
+        app.typeKey("n", modifierFlags: [.command]) // Can't use debug menu without a window
+        XCTAssertTrue(
+            resetMenuItem.waitForExistence(timeout: UITests.Timeouts.elementExistence),
+            "Reset bookmarks menu item didn't become available in a reasonable timeframe."
+        )
+        resetMenuItem.click()
+    }
+
+    /// Opens the bookmarks manager via the menu
+    static func openBookmarksManager(app: XCUIApplication, manageBookmarksMenuItem: XCUIElement) {
+        XCTAssertTrue(
+            manageBookmarksMenuItem.waitForExistence(timeout: UITests.Timeouts.elementExistence),
+            "Manage bookmarks menu item didn't become available in a reasonable timeframe."
+        )
+        manageBookmarksMenuItem.click()
+    }
+
+    /// Open the initial site to be bookmarked, bookmarking it and/or escaping out of the dialog only if needed
+    /// - Parameter url: The URL we will use to load the bookmark
+    /// - Parameter pageTitle: The page title that would become the bookmark name
+    /// - Parameter bookmarkingViaDialog: open bookmark dialog, adding bookmark
+    /// - Parameter escapingDialog: `esc` key to leave dialog
+    /// - Parameter folderName: The name of the folder where you want to save the bookmark. If the folder does not exist, it fails.
```

Comment:
Thanks a lot for taking time to document the code, especially for these utility APIs 🙏